src/coreclr/tools/Common/TypeSystem/Ecma/EcmaSignatureParser.cs

```diff
@@ -169,10 +169,16 @@ private TypeDesc ParseTypeImpl(SignatureTypeCode typeCode)
 
                             var lowerBoundsCount = _reader.ReadCompressedInteger();
                             int []lowerBounds = lowerBoundsCount > 0 ? new int[lowerBoundsCount] : Array.Empty<int>();
+                            bool nonZeroLowerBounds = false;
                             for (int j = 0; j < lowerBoundsCount; j++)
-                                lowerBounds[j] = _reader.ReadCompressedSignedInteger();
+                            {
+                                int loBound = _reader.ReadCompressedSignedInteger();
+                                if (loBound != 0)
+                                    nonZeroLowerBounds = true;
+                                lowerBounds[j] = loBound;
+                            }
 
-                            if (boundsCount != 0 || lowerBoundsCount != 0)
+                            if (boundsCount != 0 || lowerBoundsCount != rank || nonZeroLowerBounds)
```

Comment:
Is there a reason why this is `lowerBoundsCount != rank || nonZeroLowerBounds` instead of just `nonZeroLowerBounds`? It seems natural that if we treat `[0..,0..]` same as `[,]`, we should also treat `[0..,]` the same.

Reply:
We don't want to treat [,] the same as [0...,0...] those are considered incompatible in a signature. Only [0...,0...] would be a valid scenario for a rank 2 mdarray to not carry the extra signature data.